src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -134,49 +91,71 @@ def test_WINNF_FT_S_FAD_1(self):
-                                      data['recordData'][0]['ppaInfo']['ppaExpirationDate'])                                    
\ No newline at end of file
+        for activity_dump_file in response['files']:
+            # Verify the record type
+            # Verify the response files with ActivityDumpFile.schema.json Object schema
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json", activity_dump_file)
+            # Get json data from url
+            data = self._sas.DownloadFile(activity_dump_file['url'])
+            if activity_dump_file['recordType'] == 'cbsd':
+                 # Verify that everything in the full dump matches a cbsd or grant created at the beginning
+                for record in data['recordData']:
+                # Verify the response files with CbsdData.schema.json Object schema
+                    self.assertContainsRequiredFields("CbsdData.schema.json", record)
+                    self.assertTrue(record['registration']['fccId'] in (device_a['fccId'], device_c['fccId']))
+                # Verify all the previous activities on CBSDs and Grants exist in the dump files    
+                for index, device in enumerate([device_a, device_c]):
+                    record_id = 'cbsd/'+ device['fccId']+'/'+ hashlib.sha1(device['cbsdSerialNumber']).hexdigest()
+                    cbsd_record = [record['registration'] for record in data['recordData'] if record['id'] == record_id]
+                    self.assertEqual(first_cbsd['cbsdCategory'],second_cbsd['cbsdCategory'])
+                    self.assertEqual(device['airInterface']['radioTechnology'],\
+                                     cbsd_record['airInterface']['radioTechnology'])
+                    self.assertEqual(device['installationParam']['latitude'],\
+                                     cbsd_record['installationParam']['latitude'])
+                    self.assertEqual(device['installationParam']['longitude'],\
+                                     cbsd_record['installationParam']['longitude'])
+                    self.assertEqual(device['installationParam']['height'],\
+                                     cbsd_record['installationParam']['height'])
+                    self.assertEqual(device['installationParam']['heightType'], \
+                                     cbsd_record['installationParam']['heightType'])
+                    self.assertEqual(device['installationParam']['indoorDeployment'], \
+                                     cbsd_record['installationParam']['indoorDeployment'])
+                    self.assertEqual(device['installationParam']['antennaGain'], \
+                                     cbsd_record['installationParam']['antennaGain'])                           
+                    # Get grants by cbsd_id
+                    grants_of_cbsd = [cbsd['grants'] for cbsd in data['recordData'] if cbsd['id'] == record_id]
```

Comment:
Similar comment to the one above (verify that the length of this list is equal to the number of grants held by this CBSD and then retrieve the element from the list).

And it looks like you're using `record['grants'][0]` below; shouldn't you be using grants_of_cbsd?